exporter_test.go

```diff
@@ -273,6 +273,25 @@ func testExporter(t *testing.T, when spec.G, it spec.S) {
 				assertReuseLayerLog(t, logHandler, "other.buildpack.id:local-reusable-layer")
 			})
 
+			when("the launch flag is in the top level table", func() {
+				it.Before(func() {
+					exporter.Buildpacks = []buildpack.GroupBuildpack{{ID: "other.buildpack.id", API: api.Buildpack.Latest().String()}}
+					fakeAppImage.AddPreviousLayer("local-non-reusable-layer-digest", "")
+					opts.OrigMetadata = platform.LayersMetadata{
+						Buildpacks: []platform.BuildpackLayersMetadata{{
+							ID:     "other.buildpack.id",
+							Layers: map[string]platform.BuildpackLayerMetadata{"local-non-reusable-layer": {LayerMetadata: platform.LayerMetadata{SHA: "local-non-reusable-layer-digest"}}},
+						}},
+					}
+				})
+				it("ignores the flag and doesn't reuse cached launch layers even if the local sha matches the sha in the metadata", func() {
```

Comment:
I believe this was originally proposed as a failure case - did it change?

Reply:
I don't think it should be a failure case. The exporter ignores flags that aren't in the "right place" (at the types table for buildpack API < 0.6, and at the top level for buildpack API >= 0.6).
Am I missing something?

Reply:
I think in practice we wouldn't get here, because it would fail in the build phase: https://github.com/buildpacks/lifecycle/pull/537/files#diff-0fd47ba02691a569b10e75c699543569da61397717979463429aa1a4a5c5b287R594

Probably it's better to be consistent, just for ease of comprehension.

Reply:
Ohh, of course that we would fail in the build phase, but I wanted to be clear that the exporter doesn't check it again and will just ignore it.
I can delete this test but I'm not sure it's the right thing to do.
I can add a comment about it.
What do you think?

Reply:
Hmm... I'd almost argue that the exporter shouldn't ignore it, and should have the same behavior WRT inconsistent data as the builder. I think we did something similar when deprecating version from the BOM - builder failed out with data that it shouldn't get because detector would have failed. IDK!

Reply:
I agree with @natalieparellano that we should redo the check. We are unlikely to get into this situation but I like the consistency.

Reply:
Just for the record - what really convinced me to redo the check is that someone can theoretically replace only our build phase but keep our export phase.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/537/commits/10e7898473c8b663ac444985d8b6fe429f57b8ba.